lib/components/NodeBaseInputField.tsx

```diff
@@ -1,101 +1,150 @@
-import { CSSProperties, memo, ReactNode, useRef, useState } from 'react'
+import { CSSProperties, ReactNode, useRef, useState } from 'react'
 import { NodeInputConfig } from '../config'
 
-type NodeBaseInputFieldProps = Pick<NodeInputConfig, 'valueType' | 'name'> & {
+export type InputHTMLTypes =
+  | 'checkbox'
+  | 'color'
+  | 'date'
+  | 'datetime-local'
+  | 'email'
+  | 'month'
+  | 'number'
+  | 'password'
+  | 'range'
+  | 'tel'
+  | 'text'
+  | 'time'
+  | 'url'
+  | 'week'
+
+type NodeBaseInputFieldProps = Pick<NodeInputConfig, 'name'> & {
+  type: InputHTMLTypes
   value: any
+  setValue: (value: any) => void
   style?: CSSProperties
   inputStyle?: CSSProperties
   onChange?: (value: any) => void
   onPointerDown?: (e: React.PointerEvent) => void
   onPointerLeave?: (e: React.PointerEvent) => void
   children?: ReactNode
+  inputMode?:
+    | 'email'
+    | 'tel'
+    | 'text'
+    | 'url'
+    | 'search'
+    | 'none'
+    | 'decimal'
+    | 'numeric'
+    | undefined
+  pattern?: string
+  maxlength?: number
+  minlength?: number
+  max?: number | string
+  min?: number | string
+  step?: number
+  placeholder?: string
 }
 
-export const NodeBaseInputField = memo(
-  ({
-    name,
-    valueType,
-    value,
-    style,
-    inputStyle,
-    onChange,
-    onPointerDown,
-    onPointerLeave,
-    children,
-  }: NodeBaseInputFieldProps) => {
-    const [_value, setValue] = useState(value)
-    const [labelVisible, setLabelVisible] = useState(true)
-    const ref = useRef<HTMLInputElement>(null)
+export const NodeBaseInputField = ({
+  name,
+  type,
+  value,
+  setValue,
+  style,
+  inputStyle,
+  onChange,
+  onPointerDown,
+  onPointerLeave,
+  inputMode,
+  pattern,
+  maxlength,
+  minlength,
+  max,
+  min,
+  step,
+  placeholder,
+  children,
+}: NodeBaseInputFieldProps) => {
```

Comment:
Code style: There are enough properties here, and possible more in the future (?) that I would probably just do `...props`, then in the input component do
```jsx
<input {...props}/>
```